zenpy/lib/endpoint.py

```diff
@@ -180,12 +180,6 @@ def __call__(self, id, **kwargs):
 
 
 class MultipleIDEndpoint(BaseEndpoint):
-    def __call__(self, *args):
```

Comment:
I'm not 100% sure....but this method is for f(1,2,3)
And the one below is for:
f(1,2,3,foo=4,bar=6)
Please correct me if I'm wrong.

Reply:
```
class A:
    def __call_(self, val):
        print(val * 1)

    def __call__(self, val):
        print(val * 2)


def main():
    a = A()
    a(1)


if __name__ == '__main__':
    main()
```
For me this prints 2, meaning that only the second __call__ is reached.

However, I _was_ wrong in assuming the first should be removed, you would need to decide which one to keep and which one to remove. It may be the second you want to keep.

Reply:
right because you hammered __call with the **same** signature.
There are *two* calls, each with a *unique* signature.  One using args and one using kwargs.  Now I'll try your experiment with both distinct signatures.  Please hold :)

Reply:
Okay, let's try the same signature from the zenpy code.
```
class A:
    def __call__(self, *args):
        print("first")

    def __call__(self, *args, **kwargs):
        print("second")


def main():
    a = A()
    a(this="makes", no="difference")
    a("the", "same", "method", "will", "be", "called")


if __name__ == '__main__':
    main()
```

This will print "second" twice.

Reply:
Yeap __call shoves the previous away and essentially shadows ANY previous declarations.  I think we should remove the first signature.  Thanks for that.

Reply:
Maybe if I used a real linter this kind of stuff would hit me over the head in an automated way :P :P :P

Reply:
PS you were missing an underbar for this particular test:

```
class A:
    def __call__(self, *args):
        print (args[0] * 1)

    def __call__(self, *args, **kwargs):
        breakpoint()
        print (kwargs['foo'] * 69)

def main():
    a = A()
    a(1, foo=1)
    a(1) # KABOOM ^_^ not an override but a shadow, de-ref error on 'foo'

if __name__ == '__main__':
    main()

```

Reply:
Administratively merging, will be bundled in next release....but if no new pressing issues come up, I'll release sooner.